### PR TITLE
Update to flow 0.120

### PR DESCRIPTION
**Pull request by danrot**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | ---
| Related issues/PRs | ---
| License | MIT
| Documentation PR | ---

#### What's in this PR?

Updates flow to version 0.120 with all required changes.

### CODE REVIEW

File: src/Sulu/Bundle/AdminBundle/Resources/js/containers/Toolbar/stores/ToolbarStore.js [hunk 1]
```diff
@@ -61,31 +62,51 @@ export default class ToolbarStore {
...
     }
 
     getItemsConfig(): Array<ToolbarItemConfig<*>> {
         return this.config.items || [];
     }
 
+    // @deprecated
     hasIconsConfig(): boolean {
+        log.warn(
+            'The "hasiconsConfig" method is deprecated since 2.1 and will be removed. ' +
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/AdminBundle/Resources/js/containers/Toolbar/stores/ToolbarStore.js:94]: `hasIconsConfig`

File: src/Sulu/Bundle/AdminBundle/Resources/js/containers/Toolbar/Toolbar.js [hunk 2]
```diff
@@ -140,23 +144,23 @@ class Toolbar extends React.Component<ToolbarProps> {
...
-                        {this.toolbarStore.hasItemsConfig() &&
+                        {!!itemsConfig && itemsConfig.length > 0 &&
                             <ToolbarComponent.Items>
                                 {this.itemsConfig.map((itemConfig, index) => getItemComponentByType(itemConfig, index))}
                             </ToolbarComponent.Items>
                         }
                     </ToolbarComponent.Controls>
                     <ToolbarComponent.Controls>
-                        {this.toolbarStore.hasIconsConfig() &&
+                        {!!iconsConfig &&
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/AdminBundle/Resources/js/containers/Toolbar/Toolbar.js:154]: missing `iconsConfig.length` check and think `!!iconsConfig` is not needed as always an empty array is now returned which is always be true?

File: src/Sulu/Bundle/AdminBundle/Resources/js/containers/Toolbar/Toolbar.js [hunk 3]
```diff
@@ -140,23 +144,23 @@ class Toolbar extends React.Component<ToolbarProps> {
                                 success={!onNavigationButtonClick && showSuccess}
                             />
                         }
-                        {this.toolbarStore.hasItemsConfig() &&
+                        {!!itemsConfig && itemsConfig.length > 0 &&
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/AdminBundle/Resources/js/containers/Toolbar/Toolbar.js:147]: think `!!itemsConfig` is not needed as always an empty array is now returned

File: src/Sulu/Bundle/AdminBundle/Resources/js/services/ResourceRequester/tests/ResourceRequester.test.js [hunk 4]
```diff
@@ -80,7 +80,7 @@ test('Should send a list get request to the correct URL with page and limit para
         limit: 5,
         page: 1,
     });
-    expect(Requester.get).toHaveBeenLastCalledWith('/snippets?page=1&limit=5&flat=false');
+    expect(Requester.get).toHaveBeenLastCalledWith('/snippets?page=1&limit=5&flat=true');
```
Comment (nnatter) [inline src/Sulu/Bundle/AdminBundle/Resources/js/services/ResourceRequester/tests/ResourceRequester.test.js:83]: is this desired behaviour? 🙈

File: src/Sulu/Bundle/AdminBundle/Resources/js/containers/Application/tests/__snapshots__/Application.test.js.snap [hunk 5]
```diff
@@ -160,7 +160,11 @@ exports[`Render based on current route 1`] = `
           </div>
           <div
             class="controls light"
-          />
+          >
+            <div
```
Comment (nnatter) [inline src/Sulu/Bundle/AdminBundle/Resources/js/containers/Application/tests/__snapshots__/Application.test.js.snap:164]: why is this rendered now although the test was not changed? 🤔

File: src/Sulu/Bundle/AdminBundle/Resources/js/services/ResourceRequester/tests/ResourceRequester.test.js (hunk 4 above)
Comment (danrot) [reply src/Sulu/Bundle/AdminBundle/Resources/js/services/ResourceRequester/tests/ResourceRequester.test.js:83]: I was thinking about it, and I would say yes. The `ResourceRequester` is used with our APIs, and they have to follow certain conventions. That includes the `flat` option to be set for retrieving lists. The same is also true for `page`, `limit`, `locale`, etc. Overriding these to other values might lead to unexpected behaviour.

**Comment by danrot** [comment]
Merged it myself after @alexander-schranz approval, so that I can merge it back and make sure it works as well in my upcoming PRs :slightly_smiling_face:

**Pull request merged**